### PR TITLE
[devicelab] reduce iterations, uninstall at end, and use  --application-binary in all startup tests

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

These devicelab tests are incredibly flaky. Currently they will build & install the same application 15 times. This causes the temp storage to fill up on android, and has a good chance of flaking on iOS due to install issues.

The change from 3 to 15 increased total test time for 2 -> 12 minutes on iOS, or roughly 5 more test equivalents for every single iOS test. Reduce the iteration count back to 5

Uninstall the app after each run so temp storage does not fill up.

### CODE REVIEW

File: dev/devicelab/lib/tasks/perf_tests.dart [hunk 1]
```diff
@@ -454,24 +454,56 @@ class StartupTest {
 
   Future<TaskResult> run() async {
     return await inDirectory<TaskResult>(testDirectory, () async {
-      final String deviceId = (await devices.workingDevice).deviceId;
-      await flutter('packages', options: <String>['get']);
-
-      const int iterations = 15;
+      final Device device = await devices.workingDevice;
+      const int iterations = 5;
```
Comment (jmagman) [inline dev/devicelab/lib/tasks/perf_tests.dart:458]: \cc @liyuqian re: https://github.com/flutter/flutter/pull/67147
Upping from 3 and then 15 significantly increased the amount of time it took this test to run.

We also suspect it's the cause for the temp storage filling up on Android (hence the uninstall at the end of every loop).

**Review by jmagman — APPROVED** [review]
LGTM
@liyuqian can comment on the concrete impact of the increased data average noice.
Even if it NEEDS to be left at 15 for some reason, the rest seems like a good improvement.

**Comment by jonahwilliams** [comment]
if it needs to be 15, then it should be restricted to one important test case. Right now this effects 11 tests, so its a large resource increase to the devicelab

**Comment by liyuqian** [comment]
All start_up tests seem to be quite noisy to me, and they're of similar importance. I think the best way forward is to reduce the iteration time so we can do a startup iteration in ~10 second so maybe we can do 50 iterations to further reduce the noise without deplete the device lab.

**Comment by jonahwilliams** [comment]
We're never going to be able to do an iteration in 10 seconds

**Comment by jmagman** [comment]
For fun I ran this change on an iPhone 6 with 15 iterations and it reduced `complex_layout_ios__start_up` from 6:47 to 43 seconds.

**Comment by jonahwilliams** [comment]
Additionally, even if our launch was 99% reliable, running it 50 times would mean that at least one run would fail about 40% of the time.

**Comment by liyuqian** [comment]
> For fun I ran this change on an iPhone 6 with 15 iterations and it reduced complex_layout_ios__start_up from 6:47 to 43 seconds.

This looks promising. If I understand it correctly, this is 2.87 seconds / iteration which is much faster than my 10 seconds / iteration goal?

> Additionally, even if our launch was 99% reliable, running it 50 times would mean that at least one run would fail about 40% of the time.

We could write the benchmark in a way to ignore a certain proportion of failures and only use the successful ones to compute the average. But as a developer, I'd really love to increase the reliability as much as possible (e.g., `flutter run` should retry certain fragile steps to make the whole command fail less often). The 1% failure may still cost a developer headaches and loss of time.

**Pull request merged**